datus/models/openai_compatible.py

```
@@ -365,6 +388,92 @@ def generate_with_json_output(self, prompt: Any, **kwargs) -> Dict:
+                return output_type.model_validate(parsed_json)
+            except (json.JSONDecodeError, Exception) as e:
+                logger.warning(f"Failed to parse structured output: {e}, content: {content[:200]}")
+                # Try to extract JSON from response
+                import re
+
+                json_match = re.search(r"\{.*\}", content, re.DOTALL)
+                if json_match:
+                    try:
+                        parsed_json = json.loads(json_match.group(0))
+                        return output_type.model_validate(parsed_json)
+                    except Exception as parse_err:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)